pkg/operator/ceph/pool/controller.go

```diff
@@ -147,6 +148,14 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 		// If not, we should wait for it to be ready
 		// This handles the case where the operator is not ready to accept Ceph command but the cluster exists
 		if !cephBlockPool.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// disable RBD stats collection if cephBlockPool is being deleted
+			cephBlockPool.Spec.EnableRBDStats = false
+			logger.Debug("configuring RBD per-image IO statistics collection")
+			if err := configureRBDStats(r.context, cephCluster.Name); err != nil {
+				return reconcile.Result{}, errors.Wrapf(err, "failed to disable stats collection for pool(s). requeing request")
```

Comment:
What is the consequence of this failing? If a non-existent pool still exists in the list for collection, will it cause anything to fail? If it will just be ignored by ceph, I would suggest we ignore this failure, log a warning, and just continue with the deletion.

Reply:
I did not see any immediate errors or warnings. But, Ceph will probably keep trying to export its stats. And, if we recreate a Pool with same name, it'll start exporting stats of this new pool.
So, it feels safer to retry? also, the command doesn't fail often so we wouldn't be re-queueing deletions frequently.